src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -90,10 +90,14 @@ protected override Expression VisitBinary(BinaryExpression node)
             {
                 var newLeft = Visit(node.Left);
                 var newRight = Visit(node.Right);
+
+                // check if the non-string expression is a null constent
+                // as this would lead to a "null.ToString()" and thus an error when executing the expression
+                var isNullConstant = new Func<Expression, bool>(xpr => xpr is ConstantExpression c && c.Value == null);
```

Comment:
This is kind of weird :) A regular method should do the job.

Reply:
ya think! It is just used in this one single method and contributes to readability

Reply:
I disagree. You can use a local method instead.

Reply:
yeah - you're right with that